src/context/router-context.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback, useEffect } from 'react'
+
+export interface Route {
+  default?: boolean
+  path?: string
+  shouldRender?(): Promise<boolean>
+  component: React.LazyExoticComponent<(...args: any[]) => React.JSX.Element | null>
+}
+
+export const RouteContext = React.createContext<{
+  // routes: Route[]
+  currentRoute: Route | undefined
+  gotoPage(route?: string): void
+}>({ currentRoute: undefined, gotoPage: () => {} })
+
+export const RouterProvider = ({ children, routes }: { children: React.ReactNode, routes: Route[] }): JSX.Element => {
```

Comment:
very basic hash-router implementation. I looked around for some and most were 200kb and unmaintained, or very large. All required some intimate knowledge or specific finagling for things we needed. 

lmk if there's something I missed.